spark/src/main/java/org/apache/iceberg/spark/ExceptionUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import java.io.IOException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.spark.sql.AnalysisException;
+
+public class ExceptionUtil {
+
+  private ExceptionUtil() {
+  }
+
+  /**
+   * Converts Checked Exceptions to Unchecked Exceptions.
+   *
+   * @param exception the exception object expected to be unchecked.
+   * @return - checked exception.
+   */
+  public static RuntimeException toUncheckedException(Throwable exception) {
```

Comment:
I'd like to have more context here, instead of just wrapping without an exception message. Can you add a string message to add to the new exception?

Reply:
This is taken care now. Additionally to add the `manifest file path` also to the exception message when doing `buildManifest`, I had to move the exception handling inside. Ref: [SparkTableUtil.java R430-R434](https://github.com/apache/iceberg/pull/1126/files#diff-531684170cfef0f1493ced85a0c8dbfcR430-R434
)
This resulted in a try-with-resources block with a reference to the writer, as the `writer.toManifestFile` should happen only after closing the writer.

```
String outputPath = FileFormat.AVRO.addExtension(location.toString());
OutputFile outputFile = io.newOutputFile(outputPath);
ManifestWriter<DataFile> writer = ManifestFiles.write(spec, outputFile);

try (ManifestWriter<DataFile> writerRef = writer) {
  fileTuples.forEachRemaining(fileTuple -> writerRef.add(fileTuple._2));
} catch (IOException e) {
  throw SparkExceptionUtil.toUncheckedException(e, "Unable to close the manifest writer: %s", outputPath);
}

ManifestFile manifestFile = writer.toManifestFile();
return ImmutableList.of(manifestFile).iterator();
```

Let me know if the above is alright. The other option I had was the below or throw from the finally block. Both looked quite odd. 
```
try {
  try {
    fileTuples.forEachRemaining(fileTuple -> writer.add(fileTuple._2));
  } finally {
    writer.close();
  }
} catch (IOException e) {
  throw SparkExceptionUtil.toUncheckedException(e, "Unable to close the manifest writer: %s", outputPath);
}
```

Reply:
I like your idea of using a ref. I've used the other pattern in some places and it's more awkward. I'll probably use the ref approach from now on.